common/Environments/Models/ComputeSystem.cs

```diff
@@ -275,6 +275,126 @@ public async Task<ComputeSystemOperationResult> ConnectAsync(string options)
         }
     }
 
+    public async Task<ComputeSystemOperationResult> PinToStartMenuAsync(string options)
+    {
+        try
+        {
+            if (_computeSystem is IComputeSystem2 computeSystem2)
+            {
+                return await computeSystem2.PinToStartMenuAsync();
+            }
+
+            var localErrorString = $"PinToStartMenuAsync for: {this} failed due to exception";
+            var ex = new InvalidOperationException(localErrorString);
+            throw ex;
+        }
+        catch (Exception ex)
+        {
+            _log.Error($"PinToStartMenuAsync for: {this} failed due to exception", ex);

```

Comment:
you probably want to format your exceptions using this style:

```Suggestion
            _log.Error(ex, $"PinToStartMenuAsync for: {this} failed due to exception");
```

Without formatting it like this your exception won't get logged. You can do this here and the other places as well

Reply:
Not sure how I got that backwards, good catch